### DISCUSSION TITLE
Using data from

### DISCUSSION THREAD

**Discussion by guidoSNT** [yamcs/yamcs#835] [Q&A] [answered]
Hi, 

I'm very new to Yamcs and I've been having some issues with packet protocols and ports. I'm using a simulator that sends data to a TCP port with a custom protocol, but it also sends information between the packets that are not related to Yamcs. As far as I've seen, GenericPacketInputStream can be used to split my custom packets based on fixed offsets from the begging of the packet, but I would like to split my packets using the header and foot. How can I read the stream and create packets by looking for these two?.

I've been thinking of creating a class that reads the port, creates the packets and then sends the stream to another so i can use the GenericPacketInputStream, but maybe there is a better option.

Any help is welcome!!

**Comment by xpromache** [comment, marked as answer]
If the size of the information in between the packets is fixed, you could just consider it part of the packet.

If it is not fixed then I'm afraid you have to make your own PacketInputStream.
